comp/metadata/packagesigning/utils/rpmutils.go

```diff
@@ -0,0 +1,173 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package utils
+
+import (
+	"bufio"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// Not const for testing purpose
+var (
+	YumConf = "/etc/yum.conf"
+	YumRepo = "/etc/yum.repos.d/"
+)
+
+const (
+	dnfConf  = "/etc/dnf/dnf.conf"
+	zyppConf = "/etc/zypp/zypp.conf"
+	zyppRepo = "/etc/zypp/repos.d/"
+)
+
+// getMainGPGCheck returns gpgcheck and repo_gpgcheck setting for [main] table
+func getMainGPGCheck(pkgManager string) (bool, bool) {
+	repoConfig, _ := GetRepoPathFromPkgManager(pkgManager)
+	if repoConfig == "" {
+		// if we end up in a non supported distribution
+		return false, false
+	}
+	mainConf, _ := ParseRepoFile(repoConfig, MainData{})
+	return mainConf.Gpgcheck, mainConf.RepoGpgcheck
+}
+
+// GetRepoPathFromPkgManager returns the path to the configuration file and the path to the repository files for RH or SUSE based OS
+func GetRepoPathFromPkgManager(pkgManager string) (string, string) {
+	if pkgManager == "yum" {
+		return YumConf, YumRepo
+	} else if pkgManager == "dnf" {
+		return dnfConf, YumRepo
+	} else if pkgManager == "zypper" {
+		return zyppConf, zyppRepo
+	}
+	return "", ""
+}
+
+// Repositories is a struct to store the repo name
+type Repositories struct {
+	RepoName string `json:"repo_name"`
+}
+
+// MainData contains the global definitions of gpg checks
+type MainData struct {
+	Gpgcheck         bool
+	LocalpkgGpgcheck bool
+	RepoGpgcheck     bool
+}
+type repoData struct {
+	baseurl      []string
+	enabled      bool
+	metalink     string
+	mirrorlist   string
+	gpgcheck     bool
+	repoGpgcheck bool
+	gpgkey       []string
+}
+
+type multiLine struct {
+	inside bool
+	name   string
+}
+
+// ParseRepoFile extracts information from yum repo files
+// Save the global gpgcheck value when encountering a [main] table (should only occur on `/etc/yum.conf`)
+// Match several entries in gpgkey field, either file references (file://) or http(s)://. From observations,
+// these reference can be separated either by space or by new line. We assume it possible to mix file and http references
+func ParseRepoFile(inputFile string, mainConf MainData) (MainData, map[string][]Repositories) {
```

Comment:
Based on the comment, this is meant to work only from Yum repo files. Should we modify the function name to better reflect that? `ParseYumRepoFiles` ?

Reply:
I can do that